indico/modules/rb/client/js/common/rooms/edit/RoomEditOptions.jsx

```diff
@@ -0,0 +1,132 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React, {useMemo} from 'react';
+import {Dropdown, Form, Header, Tab} from 'semantic-ui-react';
+import _ from 'lodash';
+import {FieldArray} from 'react-final-form-arrays';
+import PropTypes from 'prop-types';
+import {Translate} from 'indico/react/i18n';
+import {FinalCheckbox, FinalField, FinalInput} from 'indico/react/forms';
+import EquipmentList from './EquipmentList';
+import DailyAvailability from './DailyAvailability';
+import NonBookablePeriods from './NonBookablePeriods';
+
+export default function RoomEditOptions({active, showEquipment, globalAttributes}) {
+  const attributeTitles = useMemo(
+    () => (globalAttributes ? _.fromPairs(globalAttributes.map(x => [x.name, x.title])) : []),
```

Comment:
can `globalAttributes` end up being `null` or `undefined`? If not, the default value for it is `[]` thus you don't need `?` operator at all.

Reply:
Yes, the default value is `[]` but I always expect the error case where `globalAttributes` ends up being null.

Reply:
Hm.. then, in case of an invalid value for the component's prop, should we really account for that? This seems to me like it should never happen and we should just let any expection caused by the invalid value to propagate.

Reply:
The problem is, if by any reason the `globalAttributes` request fails we have a null value that is not being correctly handled here (will crash). It is not a default prop, so I think we should always keep this as defensive.

Reply:
Ahh.. I see now. Then yeah, it is perfectly fine to account for that ;)